src/main/java/carpet/fakes/BlockPistonBehaviourInterface.java

```diff
@@ -0,0 +1,28 @@
+package carpet.fakes;
+
+import carpet.mixins.PistonStructureResolver_customStickyMixin;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.piston.PistonStructureResolver;
+import net.minecraft.world.level.block.state.BlockState;
+
+/**
+ * Opt-in Interface that allows for more control on a blocks interaction within the {@link PistonStructureResolver} via {@link PistonStructureResolver_customStickyMixin}
+ */
+public interface BlockPistonBehaviourInterface {
+
+    /**
+     * @return whether this block is sticky in any way when moved by pistons
+     */
+    default boolean isSticky(BlockState state){
```

Comment:
```suggestion
    default boolean isSticky(BlockState state) {
```
Ik it was like this in the old iface, but good opportunity to fix it given you're changing it.

Reply:
Done and did such for the bottom method also